cfspopcon/file_io.py

```diff
@@ -121,5 +121,16 @@ def write_point_to_file(dataset: xr.Dataset, point_key: str, point_params: dict,
 
     output_dir.mkdir(parents=True, exist_ok=True)
 
+    class RoundingFloat(float):
+        """A formatter to control how floats are written to JSON.
+
+        From: https://stackoverflow.com/questions/54370322/how-to-limit-the-number-of-float-digits-jsonencoder-produces
+        """
+
+        __repr__ = staticmethod(lambda x: f"{x:#.10g}")
+
+    json.encoder.c_make_encoder = None  # type:ignore[attr-defined]
+    json.encoder.float = RoundingFloat  # type:ignore[attr-defined]
```

Comment:
The problem I see is that this globally modifies the `json` behaviour for everything. This could be unexpected for users that use `cfspopcon`.

Would it work if we moved this to the top of `tests/conftest.py` such that `json` would only get modified like this in the tests where we care about the regression results?

Reply:
In https://stackoverflow.com/questions/54370322/how-to-limit-the-number-of-float-digits-jsonencoder-produces there's an example showing how to subclass the `json.JSONEncoder` which could potentially be an option

Reply:
That does seem to require implementing quite a bit of logic though. 
What's the downside of only doing this patch in the testing?

Reply:
I'm not sure if `conftest.py` will get picked up by `poetry run python tests/utils/regression_results.py`. It's run in a separate stage to `pytest`.

I could instead make a copy of the default JSON encoder, and then modify it with the above changes.

Reply:
OK, after digging for a bit, I figured I could do this with a ContextManager. The following test shows that the changes are reversible.

```
import json

test_dict = dict(
    a = 1234567891011.121314,
    b = [1234567891011.121314],
    c = dict(
        a2 = 1234567891011.121314,
        b2 = [1234567891011.121314],
    )
)

class RoundingFloat(float):
    """A formatter to control how floats are written to JSON.

    From: https://stackoverflow.com/questions/54370322/how-to-limit-the-number-of-float-digits-jsonencoder-produces
    """

    __repr__ = staticmethod(lambda x: f"{x:#.10g}")  # type:ignore[assignment,unused-ignore]

class ModifyFloatRepr:

    def __enter__(self):
        self.c_make_encoder = json.encoder.c_make_encoder
        json.encoder.c_make_encoder = None
        json.encoder.float = RoundingFloat
        return self

    def __exit__(self, exc_type, exc_value, traceback):
        json.encoder.c_make_encoder = self.c_make_encoder
        json.encoder.float = float

with open(f"original_flavour.json", "w") as file:
    json.dump(test_dict, file, indent=4, sort_keys=True)

with ModifyFloatRepr():
    with open(f"new_and_improved.json", "w") as file:
        json.dump(test_dict, file, indent=4, sort_keys=True)

with open(f"original_flavour_v2.json", "w") as file:
    json.dump(test_dict, file, indent=4, sort_keys=True)

with ModifyFloatRepr():
    with open(f"new_and_improved_v2.json", "w") as file:
        json.dump(test_dict, file, indent=4, sort_keys=True)
```